framework/src/consumption/lib/redshift/redshift-serverless/redshift-serverless-workgroup.ts

```diff
@@ -190,6 +195,34 @@ export class RedshiftServerlessWorkgroup extends TrackedConstruct implements ICo
     return this.redshiftData;
   }
 
+  /**
+   * Creates an instance of `RedshiftDataSharing` to manage the workgroup's data sharing.
+   * @param createVpcEndpoint if set to true, create interface VPC endpoint for Redshift Data API
+   * @param existingInterfaceVPCEndpointId if `createVpcEndpoint` is false, and if this is populated,
+   * then the Lambda function's security group would be added in the existing VPC endpoint's security group.
+   * @returns `RedshiftDataSharing`
+   */
+  public dataSharing(id: string, createVpcEndpoint?: boolean, existingInterfaceVPCEndpointId?: string): RedshiftDataSharing {
+    if (!this.redshiftDataSharing) {
+      const dataAccess = this.accessData(`${id}-dataAccess`, createVpcEndpoint, existingInterfaceVPCEndpointId);
```

Comment:
Why do you need another accessData?

Reply:
I'm reusing the same data-access, but this is needed to execute the grant commands and also the create database from share command